crates/cext/src/circuit.rs

```diff
@@ -378,6 +381,59 @@ pub struct OpCounts {
     len: usize,
 }
 
+/// @ingroup QkCircuit
+/// Append an arbitrary unitary matrix to the circuit.
+/// The user passes a row-major array of interleaved real-imaginary pairs #[
+/// representing the unitary matrix.
+///
+/// @param curcuit      A pointer to the circuit to add the unitary to.
+/// @param num_qubits   The width of unitary gate.
+/// @param qubits       The pointer to array of quibit indices.
+/// The array length must be equal to num_qubits.
+/// @param matrix       The pointer to the row-major `(re, im)` doucle array///
+/// # Example
+///
+///
+///
```

Comment:
CI failed because you're required to have a safety section in the docstring for an unsafe function that documents the assumptions around safe usage for this function. This is important because there are a lot of assumptions around correct usage baked into the function and we need to document them so people know how to correctly use this function in C.